lib/web_ui/test/canvaskit/fallback_fonts_golden_test.dart

```diff
@@ -123,6 +123,102 @@ void testMain() {
       // TODO: https://github.com/flutter/flutter/issues/71520
     }, skip: isIosSafari || isFirefox);
 
+    test('will put the Noto Emoji font before other fallback fonts in the list',
+        () async {
+      Completer<void> fontChangeCompleter = Completer<void>();
+      // Intercept the system font change message.
+      ui.window.onPlatformMessage = (String name, ByteData? data,
```

Comment:
This seems a bit complicated. Can we use `notoDownloadQueue.debugWhenIdle()` instead? Maybe we can also use `isPending` and `debugIsLoadingFonts` to make sure the download has been initiated.

Reply:
Done. This cleans up a lot of copy-pasted code in this test